symbulate/tests/test_distributions.py

```diff
@@ -17,18 +17,460 @@ def test_p_one(self):
     def test_sum(self):
         X = RV(Bernoulli(.4) ** 5)
         sims = X.apply(sum).sim(Nsim)
+        key_list = [keys for keys in sims.tabulate()]
         obs = list(sims.tabulate().values())
-        exp = Nsim * stats.binom(n=5, p=.4).pmf(range(6))
+        exp = [Nsim * stats.binom(n = 5, p = .4).pmf(k) for k in key_list]
+        pval = stats.chisquare(obs, exp).pvalue
+        self.assertTrue(pval > 0.01)
+
+    def test_Bernoulli_Binomial_n_1(self):
+        exp, obs, new_key = [], [], []    
+        X = RV(Bernoulli(.4))
+        sims = X.sim(Nsim)
+        key_list = list(sims.tabulate().keys())
+        for k in key_list:
+            outcome = Nsim * stats.binom(n = 1, p = .4).pmf(k)
+            if outcome > 5:
+                exp.append(outcome)
+                new_key.append(k)
+        obs = [sims.tabulate()[k] for k in new_key]        
```

Comment:
This is really inefficient. Please call `sims.tabulate()` once, store the results, and append to `obs` inside the  `for` loop. There's no point of having `new_key`.

Please fix this everywhere below.